cibyl/sources/elasticsearch/api.py

```diff
@@ -355,6 +365,116 @@ def get_deployment(self, **kwargs):
+                "bool": {
+                    "must": [
+                        {},
+                        {
+                            "bool": {
+                              "should": [
+                                {}
+                              ]
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_status"
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_name"
+                            }
+                        }
+                    ]
+                }
+            },
+            "sort": [{"timestamp.keyword": {"order": "desc"}}]
+        }
+
+        hits = []
+        for job in job_builds_found:
+            query_body['query']['bool']['must'][0] = {
+                "match": {
+                    "job_name.keyword": f"{job}"
+                }
+            }
+            builds = job_builds_found[job].builds
+            for build_number in builds:
+                query_body['query']['bool']['must'][1]['bool']['should'] = {
+                    "match": {
+                        "build_num": build_number
+                    }
+                }
+                results = self.__query_get_hits(
+                    query=query_body,
+                    index='logstash_jenkins'
+                )
+
+                # We need to process all the results because
+                # in the same build we can have different tests
+                # So we can't use '"size": 1' in the query
+                if results:
+                    for result in results:
+                        hits.append(result)
+
+        for hit in hits:
+            job_name = hit['_source']['job_name']
+            build_number = hit['_source']['build_num']
+            test_name = hit['_source']['test_name']
+            test_status = hit['_source']['test_status']
+            class_name = hit['_source'].get(
+                'test_class_name',
+                None
+            )
+            # Some build is not parsed good and contains
+            # More info than a time in the field
+            try:
+                test_duration = hit['_source'].get(
+                    'test_time',
+                    None
+                )
+                if test_duration:
+                    test_duration = float(test_duration)*1000
+            except ValueError:
+                LOG.debug("'test_time' field is not well parsed in "
+                          "elasticsearch for job: %s and build ID: %s",
+                          job_name,
+                          build_number
+                          )
+                continue
+
+            build_object = Build(
+                build_id=build_number
+            )
+
+            job_builds_found[job_name].add_build(build_object)
```

Comment:
I tested it
`$ cibyl --jobs DFG-network-networking-ovn-17.0_director-rhel-virthost-3cont_2comp_1ipa-ipv4-vlan-tls --tests --last-build
INFO     cibyl.sources.elasticsearch.client Connection established successfully with elasticsearch instance: http://seal52.qa.lab.tlv.redhat.com:9200
INFO     cibyl.plugins        Loading plugin: openstack
INFO     cibyl.orchestrator   Took 8.00s to query system osp_jenkins using source elasticsearch of type elasticsearch using method get_tests
Environment: osp_phases
  System: osp_jenkins
      Job: DFG-network-networking-ovn-17.0_director-rhel-virthost-3cont_2comp_1ipa-ipv4-vlan-tls
        Build: 13
          Status: FAILURE
        Build: 13
          Test: test_create_subnet_when_quotas_is_full[id-0b7f99e3-9f77-45ce-9a89-b39a184de618,negative]
            Result: SUCCESS
`
This output shows Build 13 twice.
I think this line triggers that, can you review it?

Reply:
Change done: https://github.com/rhos-infra/cibyl/pull/189#discussion_r862707186